### PR TITLE
Article/credential suppression

### PR DESCRIPTION
**Pull request by bcw1000**
Please think about exposing our own security posture - this blog is about that.  For that reason, I did not include any code or details of our suppression policy, or the plumbing to apply it.

Add more?  What?

### CODE REVIEW

File: _posts/2023-6-20-credential-suppression.markdown [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: Leaked Credentials!
+date: 2023-6-20
```
Comment (ConnectivityChris) [inline _posts/2023-6-20-credential-suppression.markdown:4]: Once we are ready to commit, will need to update this to the day

File: _posts/2023-6-20-credential-suppression.markdown [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+layout: post
```
Comment (ConnectivityChris) [inline _posts/2023-6-20-credential-suppression.markdown:2]: We have a new layout `postv2` which is a bit more modern, shouldn't effect content, just visual

File: _posts/2023-6-20-credential-suppression.markdown [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+title: Leaked Credentials!
+date: 2023-6-20
+tags: aws account control tower sso organizations credentials permission policy
+author: Bret Comstock Waldow
+---
+# Summary
+This is a discussion of issues due to leaked AWS credentials and suggestions of how to avoid these problems.
+# AWS Credentials & types
+AWS credentials permit a user to assume an AWS identity.  Depending on the type of AWS identity, the identity may also have inherent or assigned permission to acess AWS resources and initiate actions with them.
```
Comment (ConnectivityChris) [inline _posts/2023-6-20-credential-suppression.markdown:11]: `acess` > `access`

File: _posts/2023-6-20-credential-suppression.markdown [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+SSO credentials, whether for Console access or for CLI use, have temporary lifespans.  This lifespan is long enough for serious mischief to occur if leaked to a hostile party, or even a less-competent one.
+
+While these credentials expire, during their lifespan, they are independent and AWS offers no way to cancel them.  However, a clause in a well-targeted policy leads to them being ignored, which may be just as effective.
+## SCP approach
+Control Tower is run from an initial AWS account and designates this account to be the management account.  At the same time, an Organization is created, as well as several supporting accounts.  Control Tower also creates Permission Sets to apply to accounts in the Organization.
+
+Apart from the management account, all the created accounts and any invited to join the Organization are in the Organization, which provides a mechanism to apply an AWS Service Control Policy (SCP) to any member of the Organization, including the root.  An SCP applied to the root will be in force on any account or Organizational Unit (OU) in the Organization, and in this way we may apply a blanket policy that applies to all but the management account.
```
Comment (Thashin) [inline _posts/2023-6-20-credential-suppression.markdown:73]: I think the below suggestion makes sense

```suggestion
Apart from the management account, all the created accounts and any invited to join the Organization are in the Organization, which provides a mechanism to apply an AWS Service Control Policy (SCP) to any member of the Organization, including the root.  An SCP applied to the root will be enforced on any account or Organizational Unit (OU) in the Organization, and in this way we may apply a blanket policy that applies to all but the management account.
```

**Comment by Hamo69** [comment]
Bret are you going to publish this? Or just leave it? If the latter, please close

**Comment by bcw1000** [comment]
> Bret are you going to publish this? Or just leave it? If the latter, please close

No one has approved it.

I'd also like to hear if it exposes too much of our security posture, or doesn't say enough to be a useful blog, in which case I could revise it.

File: _posts/2023-6-20-credential-suppression.markdown (hunk 1 above)
Comment (Hamo69) [reply _posts/2023-6-20-credential-suppression.markdown:4]: agreed

File: _posts/2023-6-20-credential-suppression.markdown [hunk 5]
```diff
@@ -0,0 +1,137 @@
+---
+layout: postv2
+title: Leaking Credentials!
+date: 2023-6-20
+tags: aws account control tower sso organizations credentials permission policy
+author: Bret Comstock Waldow
```
Comment (Hamo69) [inline _posts/2023-6-20-credential-suppression.markdown:6]: Also add an image, so when it is posted, the image is visible in socials

File: _posts/2023-6-20-credential-suppression.markdown [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+---
+# Summary
+This is a discussion of issues due to leaked AWS credentials and suggestions of how to avoid these problems.
```
Comment (Hamo69) [inline _posts/2023-6-20-credential-suppression.markdown:9]: Is it a discussion - or a post/article?

Comment (Hamo69) [reply _posts/2023-6-20-credential-suppression.markdown:9]: Probably needs more bang in the opening sentence

**Review by Hamo69 — APPROVED** [review]
Approved - just needs a few things updated

**Pull request merged**